gossipsub.go

```diff
@@ -70,20 +70,37 @@ var (
 
 	// Number of peers to opportunistically graft
 	GossipSubOpportunisticGraftPeers = 2
+
+	// If a GRAFT comes before GossipSubGraftFloodThreshold has ellapsed since the last PRUNE,
+	// then there is no PRUNE response emitted. This protects against GRAFT floods and should be
+	// less than GossipSubPruneBackoff.
+	GossipSubGraftFloodThreshold = 10 * time.Second
+
+	// Maximum number of messages to include in an IHAVE message. Also controls the maximum
+	// number of IHAVE ids we will accept and request with IWANT from a peer within a heartbeat,
+	// to protect from IHAVE floods. You should adjust this value from the default if your
+	// system is pushing more than 5000 messages in GossipSubHistoryGossip heartbeats; with the
+	// defaults this is 1666 messages/s.
+	GossipSubMaxIHaveLength = 5000
```

Comment:
No, we are not tracking in flight IWANTs. 
I agree that we do want to track them, and it will make it much easier to respond to distributed IHAVE floods as well.
I'll open an issue so that we can tackle this, as it is rather complex and I want to think a bit about it.

Reply:
follow up in #285.